File: src/167/ Two Sum II - Input Array Is Sorted/Solution.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] twoSum(int[] numbers, int target) {
        Map<Integer, Integer> store = new HashMap<>();
        for (int i = 0; i < numbers.length; i++) {
            int num = numbers[i];
            if (store.containsKey(target - num)) {
                return new int[]{store.get(target - num) + 1, i + 1};
            }
            store.put(num, i);
        }
        return new int[]{};
    }
}

File: src/167/ Two Sum II - Input Array Is Sorted/Solution.kt
package `167`.` Two Sum II - Input Array Is Sorted`

class Solution {
    fun twoSum(numbers: IntArray, target: Int): IntArray {

        numbers.forEachIndexed { i, num ->
            val v = numbers.binarySearch(target - num, i + 1)
            if (v > 0) {
                return intArrayOf(i + 1, v + 1)
            }
        }
        return intArrayOf()
    }
}